lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -240,12 +298,18 @@ static int trace_init(void)
 {
 	int err;
 
+	if (!trace_backend.init ||
+	    !trace_backend.deinit ||
+	    !trace_backend.write) {
+		LOG_ERR("trace backend does not support init, deinit and write");
```

Comment:
This code prevents `NULL` dereference, but it can still be improved. I know that clustering these checks looks tidy, but I would separate them. Reading the current log it would appear that if, for example, I forgot to implement deinit, the message will tell me that my backend does not support also init and deinit.
I would also go for "implement" rather than "support" since the interface is implemented.
If you want to keep it clustered, then the log message should say "or" rather than "and", but it still wouldn't be clear for the user which one is not implemented.

Reply:
Ok, I will update this to `trace backend must implement init, deinit and write`. It should be easy to check which one is not implemented in the code anyway, so I would rather keep it grouped and small than have three different checks. This is not something that change from backend to backend as the others (read, clear, ...).

Reply:
That's also ok.